config/params/mainnet_config.go

```diff
@@ -165,6 +165,7 @@ var mainnetBeaconConfig = &BeaconChainConfig{
 	DomainSyncCommittee:               bytesutil.ToBytes4(bytesutil.Bytes4(7)),
 	DomainSyncCommitteeSelectionProof: bytesutil.ToBytes4(bytesutil.Bytes4(8)),
 	DomainContributionAndProof:        bytesutil.ToBytes4(bytesutil.Bytes4(9)),
+	DomainApplicationMask:             [4]byte{0, 0, 0, 1},
```

Comment:
Another option is `bytesutil.ToBytes4(common.HexToHash("0x00000001").Bytes()[28:])` but that's obvious uglier. Yours is good!

Reply:
This is a drop in readability imo, why can't we follow the previous pattern ?
```
bytesutil.ToBytes4(bytesutil.Bytes4(X)),
```

Reply:
I see three options to this:

1.) bytesutil.ToBytes4(bytesutil.Bytes4(16777216))
2.) bytesutil.ToBytes4(common.HexToHash("0x00000001").Bytes()[28:])
3.) [4]byte{0, 0, 0, 1},

If we compare all three with the spec definition `DomainType('0x00000001'), 2 or 3 is the closest. One which we hard a magic number is not pretty, and it's less auditable. Any options that I may have missed?

Reply:
My main concern is just reading how the config is initialized and trying to know why `[4]byte{0, 0, 0, 1}` was written the way it is. It isn't obvious until I read the spec again. Either you can add a comment stating that it is intended to be a mask which is why it was initialized that way. Or you can try doing `bytesutil.ToBytes4(bytesutil.ReverseByteOrder(bytesutil.Bytes4(1)))`. But then it is messier to read.

Reply:
I'm ok with either option. Will let @saolyn decide

Reply:
@nisdas, @terencechain, all of the proposed solutions were a bit messy.
As an alternative, I decided to create a new function `Uint32ToBytes4` that will do the same thing as `bytesutil.ToBytes4(bytesutil.Bytes4(N))` but will also work for other values. 
This way it should be a lot cleaner and better represent the actual values from the spec.

Please let me know what you think.